src/rgw/rgw_rest_s3.cc

```diff
@@ -6224,18 +6224,33 @@ int RGWSelectObj_ObjStore_S3::send_response_data(bufferlist& bl, off_t ofs, off_
     dump_errno(s);
   }
 
+  auto bl_len = bl.get_num_buffers();
+
   // Explicitly use chunked transfer encoding so that we can stream the result
   // to the user without having to wait for the full length of it.
   if (chunk_number == 0) {
     end_header(s, this, "application/xml", CHUNKED_TRANSFER_ENCODING);
   }
 
   int status=0;
+  int i=0;
+
   for(auto& it : bl.buffers()) {
+
+    ldout(s->cct, 10) << "processing segment " << i << " out of " << bl_len << " off " << ofs
+                      << " len " << len << " obj-size " << s->obj_size << dendl;
+
+    if(it.length() == 0) {
+      ldout(s->cct, 10) << "s3select:it->_len is zero. segment " << i << " out of " << bl_len
+                        <<  " obj-size " << s->obj_size << dendl;
+      return 0;
```

Comment:
this should `continue` instead of `return`, so we can still process any segments that come after

Reply:
BTW, pet some runs I did, it create only single segment per chunk.
I guess there are cases where it could be more than one segment.